### ISSUE TITLE
Add support for custom AssemblyName property to AssemblyGenerator class

### ISSUE THREAD

**Issue by pwhe23** [JasperFx/lamar#99]
I would like an assembly that is generated by Lamar to be able to access internals in another assembly.  This is a simple LinqPad example of what I was trying to accomplish:

```
void Main()
{
    var generator = new AssemblyGenerator();
    generator.ReferenceAssembly(typeof(Test).Assembly);

    var assembly = generator.Generate(x =>
    {
        x.Namespace("Generated");
        x.StartClass("CallTest");

        x.Write("BLOCK:public string SayHello()");
        x.Write("return new UserQuery.Test().Hello();");
        x.FinishBlock();

        x.FinishBlock();  // Finish the class
        x.FinishBlock();  // Finish the namespace
    });
    
    var callTestType = assembly.ExportedTypes.Single();
    var callTestInstance = Activator.CreateInstance(callTestType);
    var result = callTestType.GetMethod("SayHello").Invoke(callTestInstance, new object[0]);
    Console.WriteLine(result);
}

internal class Test
{
    internal string Hello()
    {
        return "World";
    }
}
```

I can use the InternalsVisibleTo assembly attribute but it requires a consistent assembly name. I think adding an AssemblyName string property to the AssemblyGenerator class to use if provided instead of GetRandomFileName (which could still be the default) would work for my case.

I went ahead and forked Lamar and created a branch to accomplish this with a couple of tests. I tested it locally for my use case and it worked perfectly with InternalsVisibleTo. I would be glad to create a Pull Request if desired. Here is a working LinqPad example of the end result:

```
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Generated")]
void Main()
{
    var generator = new AssemblyGenerator { AssemblyName = "Generated" };
    generator.ReferenceAssembly(typeof(Test).Assembly);

    var assembly = generator.Generate(x =>
    {
        x.Namespace("Generated");
        x.StartClass("CallTest");

        x.Write("BLOCK:public string SayHello()");
        x.Write("return new UserQuery.Test().Hello();");
        x.FinishBlock();

        x.FinishBlock();  // Finish the class
        x.FinishBlock();  // Finish the namespace
    });
    
    var callTestType = assembly.ExportedTypes.Single();
    var callTestInstance = Activator.CreateInstance(callTestType);
    var result = callTestType.GetMethod("SayHello").Invoke(callTestInstance, new object[0]);
    Console.WriteLine(result);
}

internal class Test
{
    internal string Hello()
    {
        return "World";
    }
}

```

https://github.com/pwhe23/lamar/tree/assembly-name

I don't have much experience with open source, so I hope I did everything correctly. This is a very cool project, I've been following your work for a while and am glad for an opportunity to give a little back, thanks!

**Comment by jeremydmiller** [comment]
I'd take a pull request for that, and thank you. Isn't something I would have thought to do.

**Comment by jeremydmiller** [comment]
This was closed by #101
